src/dvc_objects/fs/base.py

```diff
@@ -378,8 +378,13 @@ def rmdir(self, path: AnyFSPath) -> None:
     def rm_file(self, path: AnyFSPath) -> None:
         self.fs.rm_file(path)
 
-    def rm(self, path: AnyFSPath) -> None:
-        self.fs.rm(path, recursive=True)
```

Comment:
Was there a reason to hardcode recursive=True here?

It doesn't comply with fsspec and was introducing even more overhead in the underlying `expand_path` call for each `dvc-{x}` filesystem

Reply:
I'm not actually sure, given that we also define `LocalFilesystem.rm(remove=False)`

I think we should be exposing the flag properly here and defaulting to false

Reply:
The reason is that our old filesystem's (aka remote aka tree) `rm` method was always recursive. Might be not needed anymore.

Reply:
I checked `dvc` repo and the only place potentially affected I have found is in `output` (https://github.com/iterative/dvc/pull/8825)

Reply:
@daavoo Thank you for checking it!